pkg/compactor/compactor.go

```diff
@@ -733,6 +733,11 @@ func (c *Compactor) compactUserWithRetries(ctx context.Context, userID string) e
 
 func (c *Compactor) compactUser(ctx context.Context, userID string) error {
 	bucket := bucket.NewUserBucketClient(userID, c.bucketClient, c.cfgProvider)
+
+	if ib, ok := bucket.WithExpectedErrs(bucket.IsObjNotFoundErr).(objstore.InstrumentedBucket); ok {
+		bucket = ib
```

Comment:
This would ignore all `ObjNotFoundErr` error, right? For normal 404 errors I think we still want to have a way to catch those.

Reply:
Im not sure if we wanna get 404 in this metric at all.
I can be convinced otherwise but right now we have many 404 on the compactor codepath that we just ignore but is still in this metric.

Reply:
`thanos_objstore_bucket_operation_failures_total` it mostly fails on which operation? If the 404 is caused by some special objects then we can probably just ignore those

Reply:
We fail every time we try to get any marker and the marker is not there (ex: the skip_compaction marker)

Reply:
This sounds like a bug on Thanos side. No that marker is expected and shouldn't count the metric

Reply:
I still don't think this is a good way to solve the issue as ignoring 404 affect all objects API calls, not just the markers.

If you found the issue is related to the no compact marker, then we should have a separate issue on the Thanos side. Can you elaborate more on the cause of this issue?

Reply:
https://github.com/cortexproject/cortex/pull/4874